cmd/mdatagen/internal/templates/metrics.go.tmpl

```diff
@@ -12,13 +12,12 @@ import (
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/pcommon"
 	"go.opentelemetry.io/collector/pdata/pmetric"
-	"go.opentelemetry.io/collector/receiver"
+	{{- if or isReceiver isScraper }}
+	"go.opentelemetry.io/collector/{{ .Status.Class }}"
+	{{- end }}
 	{{- if .SemConvVersion }}
 	conventions "go.opentelemetry.io/collector/semconv/v{{ .SemConvVersion }}"
 	{{- end }}
-	{{ if .ResourceAttributes -}}
-	"go.opentelemetry.io/collector/filter"
-	{{- end }}
```

Comment:
Why was this removed?  The `filter` package is referenced in the generated code.  I now get these errors when trying to use the new mdatagen

```
receiver/tlscheckreceiver/internal/metadata/generated_metrics.go:490:44: undefined: filter
```

Reply:
https://github.com/open-telemetry/opentelemetry-collector/issues/12298